pyscript.core/src/stdlib/pyweb/pydom.py

```diff
@@ -204,6 +204,91 @@ def classes(self):
     def show_me(self):
         self._js.scrollIntoView()
 
+    def snap(
+        self,
+        to: BaseElement | str = None,
+        width: int | None = None,
+        height: int | None = None,
+    ):
+        """
+        Captures a snapshot of a video element. (Only available for video elements)
+
+        Inputs:
+
+            * to: element where to save the snapshot of the video frame to
+            * width: width of the image
+            * height: height of the image
+
+        Output:
+            (Element) canvas element where the video frame snapshot was drawn into
+        """
+        if self._js.tagName != "VIDEO":
+            raise AttributeError("Snap method is only available for video Elements")
+
+        if to is None:
+            canvas = self.create("canvas")
+            if width is None:
+                width = self._js.width
+            if height is None:
+                height = self._js.height
+            canvas._js.setAttribute("width", width)
```

Comment:
watch out attribute in canvas is a bit weird ... the accessor is usually what you want to use instead ... so `canvas._js.width = width` is probably better, or even both, with the former assigned after.

Reply:
Cool. Thanks for the hint! I'll be adding a descriptor to map Python attributes to JS attributes and that should map to accessors. Perfect fit! :)